ocaml/libs/clock/date.ml

```diff
@@ -64,12 +64,24 @@ let best_effort_iso8601_to_rfc3339 x =
       x
 
 let of_iso8601 x =
-  let rfc3339 = best_effort_iso8601_to_rfc3339 x in
-  match Ptime.of_rfc3339 rfc3339 |> Ptime.rfc3339_error_to_msg with
-  | Error _ ->
-      invalid_arg (Printf.sprintf "%s: %s" __FUNCTION__ x)
-  | Ok (t, tz, _) ->
-      {t; tz}
+  if String.length x > 5 && x.[4] <> '-' && x.[String.length x - 1] = 'Z' then
```

Comment:
It would be nice to have some comment explaining the particular reasons behind each condition